### PR TITLE
Initial code

### PR DESCRIPTION
**Pull request by otaviojava**

### CODE REVIEW

File: .platform.app.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+mounts:
+    '/.jenkins':
+        source: service
+        service: files
+        source_path: jenkins_source
+
+# The hooks executed at various points in the lifecycle of the application.
+hooks:
+    build: mkdir -p .m2 && mv settings.xml .m2/settings.xml && export MAVEN_OPTS='-Xmx64m' && wget -q http://mirrors.jenkins.io/war-stable/latest/jenkins.war
```
Comment (Crell) [inline .platform.app.yaml]: This would be much easier to read/maintain if broken to multiple lines.  There's no need to && them.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,19 @@
-# template-jenkins
-Jenkins for Platform.sh. Maintained by the Platform.sh team.
+# Jenkins template for Platform.sh
+
+This project provides a starter kit for Jenkins projects hosted on [Platform.sh](http://platform.sh).
+
+We include recommended .platform.app.yaml and .platform files that should suffice for most use cases. You are free to tweak them as needed for your particular site.
```
Comment (Crell) [inline README.md]: Backticks on the file names.

File: .platform.app.yaml (hunk 1 above)
Comment (Crell) [reply .platform.app.yaml]: For that matter, why not just put .m2 into the repository directly?  Moving the files around seems unnecessary.

File: .platform.app.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+# The name of this app. Must be unique within a project.
+name: jenkins
+
+# The runtime the application uses.
+type: "java:8"
+
+disk: 1024
+
+mounts:
```
Comment (chadwcarlson) [inline .platform.app.yaml]: How much in `.jenkins` actually needs to be writable? There's code in here, and I know @Crell has mentioned to me on other templates that we don't want to do this.

Comment (otaviojava) [reply .platform.app.yaml]: the `.jenkins` is the folder workspace for Jenkins. It creates several folders and files on it.

File: .platform/services.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+# The services of the project.
+#
+# Each service listed will be deployed
+# to power your Platform.sh project.
+
+files:
```
Comment (Crell) [inline .platform/services.yaml]: Why is this using a network file store by default?  Is there any benefit to that?

Comment (Crell) [reply .platform/services.yaml]: (There may not be much of a performance cost, but it's non-zero.)

File: .platform.app.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    '/.jenkins':
+        source: service
+        service: files
+        source_path: jenkins_source
+
+# The hooks executed at various points in the lifecycle of the application.
+hooks:
+    build: |
+      mkdir -p .m2 
+      mv settings.xml .m2/settings.xml
```
Comment (Crell) [inline .platform.app.yaml]: Why not just put .m2 in the repo and have the settings.xml file be there to begin with?

File: .platform/services.yaml (hunk 4 above)
Comment (otaviojava) [reply .platform/services.yaml]: Good question!
Is there another way to write without doing this?
https://docs.platform.sh/configuration/services/network-storage.html

Comment (otaviojava) [reply .platform/services.yaml]: I find it, thank you

File: .platform.app.yaml (hunk 3 above)
Comment (Crell) [reply .platform.app.yaml]: Are those known, that we can individually mount?  Or is this a case of "Jenkins is mixing generated code in with configuration, which is stupid, but Jenkins is just being stupid"?

Comment (otaviojava) [reply .platform.app.yaml]: Each task means a new workspace, therefore, a new folder. Plus updates, plugins and so on.

**Pull request closed without merging**